sympy/diffgeom/diffgeom.py

```diff
@@ -1,6 +1,8 @@
 from __future__ import print_function, division
 
 from itertools import permutations
+import itertools
+import copy
```

Comment:
Omg, the `copy` module. It usually portends bad things.

Reply:
In SymPy, we usually call the constructor of the class to create a copy. If the class inherits `Basic`, you can create a new object by calling `self.func( ... )`.

Subclasses of `Basic` need be immutable.